Sources/SmithyXML/Writer/Writer.swift

```diff
@@ -19,19 +21,20 @@ import struct SmithyTimestamps.TimestampFormatter
 /// This writer will write all Swift types used by Smithy models, and will also write Swift
 /// `Array` and `Dictionary` (optionally as flattened XML) given a writing closure for
 /// their enclosed data types.
```

Comment:
Modified to conform to `SmithyWriter`.

The `detach()` method is removed because nodes without content are now ignored when writing XML, eliminating the need to detach them.